### PR TITLE
Centralize `base_asset` default

### PR DESCRIPTION
**Pull request by maltekliemann**
The motivation for this pull request is [this line](https://github.com/zeitgeistpm/zeitgeist/blob/05c67d136b87167be2396977ace2aad7d48f6b9f/zrml/swaps/src/lib.rs#L1636) in `set_pool_as_stale`:

```rust
let base_asset_or_default = base_asset.unwrap_or(Asset::Ztg);
```

This makes the assumption that if `base_asset` is `None`, then it is `Asset::Ztg`. But this kind of assumption should not be made in some free function, but rather when the pool is created. Otherwise, we will keep double-booking this default value in other functions, resulting in difficult-to-maintain and error-prone code.

So, from now on, `create_pool` will set the `base_asset` to `Asset::Ztg` if it is not specified.

### CODE REVIEW

**Comment by maltekliemann** [comment]
Come to think of it: Should we just make the `base_asset` member of `Pool` non-optional? I don't see why not.

**Comment by sea212** [comment]
FYI: This requires a storage migration. We'll have to convert all `Some(x)` to `x` and all `None` to `Asset::Ztg` within the storage trie. You can have a look at the prediction markets pallet in https://github.com/zeitgeistpm/zeitgeist/pull/465 to see how it can be done. I see you have quite a lot on your plate; If desired, I can also tackle this.

**Comment by maltekliemann** [comment]
@sea212 Thanks for pointing that out and offering help! I'd like to tackle this myself. I might have questions later, though ;)

Should I just make the storage migration part of this PR?

**Comment by maltekliemann** [comment]
I've been torn all day about how many pools to test the migration with. We can either go with these two, or we can go with 60 pools that I've created.

**Comment by maltekliemann** [comment]
When performing this migration locally, I get the following strange log:

```
2022-04-08 18:20:24 Starting migration of Pool::base_asset from Optional<Asset<MarketId>> to Asset<MarketId>    
2022-04-08 18:20:24 Completed migration of Pool::base_asset from Optional<Asset<MarketId>> to Asset<MarketId>    
2022-04-08 18:20:24 ✅ no migration for System    
2022-04-08 18:20:24 ✅ no migration for Timestamp    
2022-04-08 18:20:24 ✅ no migration for RandomnessCollectiveFlip    
2022-04-08 18:20:24 ✅ no migration for Scheduler    
2022-04-08 18:20:24 ✅ no migration for Preimage    
2022-04-08 18:20:24 ✅ no migration for Balances    
2022-04-08 18:20:24 ✅ no migration for TransactionPayment    
2022-04-08 18:20:24 ✅ no migration for Treasury    
2022-04-08 18:20:24 ✅ no migration for Vesting    
2022-04-08 18:20:24 ✅ no migration for Democracy    
2022-04-08 18:20:24 ✅ no migration for AdvisoryCommittee    
2022-04-08 18:20:24 ✅ no migration for AdvisoryCommitteeMembership    
2022-04-08 18:20:24 ✅ no migration for Council    
2022-04-08 18:20:24 ✅ no migration for CouncilMembership    
2022-04-08 18:20:24 ✅ no migration for TechnicalCommittee    
2022-04-08 18:20:24 ✅ no migration for TechnicalCommitteeMembership    
2022-04-08 18:20:24 ✅ no migration for Identity    
2022-04-08 18:20:24 ✅ no migration for Sudo    
2022-04-08 18:20:24 ✅ no migration for Utility    
2022-04-08 18:20:24 ✅ no migration for Currency    
2022-04-08 18:20:24 ✅ no migration for Tokens    
2022-04-08 18:20:24 ✅ no migration for MarketCommons    
2022-04-08 18:20:24 ✅ no migration for Authorized    
2022-04-08 18:20:24 ✅ no migration for Court    
2022-04-08 18:20:24 ✅ no migration for LiquidityMining    
2022-04-08 18:20:24 ✅ no migration for RikiddoSigmoidFeeMarketEma    
2022-04-08 18:20:24 ✅ no migration for SimpleDisputes    
2022-04-08 18:20:24 ✅ no migration for Swaps    
2022-04-08 18:20:24 ✅ no migration for PredictionMarkets    
2022-04-08 18:20:24 ✅ no migration for Aura    
2022-04-08 18:20:24 ✅ no migration for Grandpa    
```

I assume that this is because we don't use the `on_runtime_upgrade` function of the swaps pallet, right?

**Comment by maltekliemann** [comment]
Other than that, the upgrade seems to go fine, although it takes a couple of blocks for `polkadot-js` to be able to access the `Pools` storage map. Is this to be expected?

**Review by sea212 — COMMENTED** [review]
Thanks, looks good overall.
I am just wondering whether we should enforce the `base_asset` parameter. It will break backwards-compatibility (requires changes in SDK and frontend), but probably helps avoiding accidentally deploying the pool with a wrong base currency.
I think since we'll have other possible currencies in the near future, this might be a good approach.

**Comment by sea212** [comment]
Please add the information that the `Pool` structure contains the `base_asset` directly instead wrapped within an Option to the changelog for devs. If you also decide to make the function parameter non-optional, please add this information to the document too. [docs/changelog_for_devs.md](https://github.com/zeitgeistpm/zeitgeist/blob/main/docs/changelog_for_devs.md).

File: zrml/swaps/src/migrations.rs [hunk 1]
```diff
@@ -1 +1,203 @@
...
+                    weights: old_pool.weights,
+                };
+                put_storage_value::<Option<Pool<BalanceOf<T>, T::MarketId>>>(
+                    SWAPS,
+                    POOLS,
+                    &key,
+                    Some(new_pool),
+                );
+                total_weight = total_weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+            }
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply zrml/swaps/src/migrations.rs:72]: Good catch, thanks! :thumbsup:

**Review by sea212 — APPROVED** [review]

**Pull request merged**